lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2434,142 +2435,39 @@ namespace {
-            Impl.ImportedDecls.end())
-          continue;
-
-        ImportedName importedName;
-        std::tie(importedName, std::ignore) = importFullName(redecl);
-        if (!importedName)
-          continue;
-
-        auto extensionDC = Impl.importDeclContextOf(
-            redecl, importedName.getEffectiveContext());
-        if (!extensionDC)
-          continue;
-
-        // We are creating an extension, so put it at the top level. This is
-        // done after creating the enum, though, because we may need the
-        // correctly nested decl context above when creating the enum.
-        while (extensionDC->getParent() &&
-               extensionDC->getContextKind() != DeclContextKind::FileUnit)
-          extensionDC = extensionDC->getParent();
-
-        auto *extension = ExtensionDecl::create(
-            Impl.SwiftContext, Impl.importSourceLoc(decl->getBeginLoc()),
-            nullptr, {}, extensionDC, nullptr, redecl);
-        enumDecl->addExtension(extension);
-        Impl.ImportedDecls[{redecl, getVersion()}] = extension;
-
-        Impl.SwiftContext.evaluator.cacheOutput(ExtendedTypeRequest{extension},
-                                                enumDecl->getDeclaredType());
-        Impl.SwiftContext.evaluator.cacheOutput(ExtendedNominalRequest{extension},
-                                                std::move(enumDecl));
-        // Keep track of what members we've already added so we don't add the
-        // same member twice. Note: we can't use "ImportedDecls" for this
-        // because we might import a decl that we don't add (for example, if it
-        // was a parameter to another decl).
-        SmallPtrSet<Decl *, 16> addedMembers;
-
-        // Insert these backwards into "namespaceDecls" so we can pop them off
-        // the end without loosing order.
-        SmallVector<clang::Decl *, 16> namespaceDecls;
-        auto addDeclsReversed = [&](auto decls) {
-          auto begin = decls.begin();
-          auto end = decls.end();
-          int currentSize = namespaceDecls.size();
-          int declCount = currentSize + std::distance(begin, end);
-          namespaceDecls.resize(declCount);
-          for (int index = declCount - 1; index >= currentSize; --index)
-            namespaceDecls[index] = *(begin++);
-        };
-        addDeclsReversed(redecl->decls());
-        while (!namespaceDecls.empty()) {
-          auto nd = dyn_cast<clang::NamedDecl>(namespaceDecls.pop_back_val());
-          // Make sure we only import the defenition of a record.
-          if (auto tagDecl = dyn_cast_or_null<clang::TagDecl>(nd))
-            // Some decls, for example ClassTemplateSpecializationDecls, won't
-            // have a definition here. That's OK.
-            nd = tagDecl->getDefinition() ? tagDecl->getDefinition() : tagDecl;
-          if (!nd)
-            continue;
+      ImportedName importedName;
+      std::tie(importedName, std::ignore) = importFullName(decl);
```

Comment:
Is there no possibility of the `correctSwiftName` being present or important?

Reply:
I don't think there's a specific reason to ignore it, it's just that compatibility wasn't considered here before.